portfolio/src/main/webapp/script.js

```diff
@@ -27,8 +27,24 @@ function addRandomFacts() {
   factContainer.innerText = fact;
 }
 
-/** Fetches the content in DataServlet and display on Home page */
+/** Fetches comments from the servers and adds them to the DOM. */
 function getData() {
-  fetch('/data').then(response => response.text()).then((data) => 
-  {document.getElementById("data-container").innerText = data;});
+  fetch('/data').then(response => response.json()).then((data) => {
+    const arrayListElement = document.getElementById('data-container');
+    arrayListElement.innerHTML = '';
+    data.forEach((comment) => {
+      arrayListElement.appendChild(createListElement(comment));
+    });
+  });
+}
+
+/** Creates an <li> element containing text. */
+function createListElement(text) {
+  const liElement = document.createElement('li');
+  liElement.innerText = text;
+  return liElement;
+}
+
+function deleteData(){
+  fetch('/delete-data');
```

Comment:
If you want to POST request, it seems that you need to specify explicitly.
```
fetch('/delete-data', {method: 'post'})
```
please refer to https://googlechrome.github.io/samples/fetch-api/fetch-post.html for details.